runscripts/animation.py

```diff
@@ -1,44 +1,86 @@
 import numpy as np
+import os
+
 from mayavi.api import Engine
 from mayavi import mlab
-# Create a new mayavi scene.
-threshold = np.load(input("enter a path to thresholded volume"))
-skeleton = np.load(input("enter a path to skeleton you want to superimpose over"))
-e = Engine()
-e.start()
-s = e.new_scene()
-mlab.contour3d(np.uint8(threshold), contours=[108.50]).actor.property.representation = 'points'
-mlab.contour3d(np.uint8(skeleton), colormap='gray')
-mlab.options.offscreen = True
-mlab.outline(s)
-# Make an animation:
-for i in range(36):
-    # Rotate the camera by 10 degrees.
-    s.scene.camera.azimuth(10)
-    # Resets the camera clipping plane so everything fits and then
-    # renders.
-    s.scene.reset_zoom()
-    # Save the scene. magnification=2 gives saves as an image when seen in fullscreen
-    mlab.savefig("anim%d.png" % i, magnification=2)
-
-# use imagemagick to create video from this frames
-# convert -set delay 20 -loop 0 -quality 1000 -scale 100% *.png /home/pranathi/animExp.mpg
-    # for coord in list(d.keys()):
-    #     x = coord[0]; y = coord[1]; z = coord[2];
-    #     mlab.text3d(x, y, z, d[coord], color=(0, 0, 0), scale=4.0)
-
-# import Image
-
-# background = Image.open("bg.png")
-# overlay = Image.open("ol.jpg")
-
-# # background = background.convert("RGBA")
-# # overlay = overlay.convert("RGBA")
-
-# new_img = Image.blend(background, overlay, 0.5)
-# new_img.save("new.png", "PNG")
-
-# fig, ax = plt.subplots(1)
-# p = ax.pcolormesh(overlay)
-# fig.colorbar(p)
 
+"""
+Create an animation using mayavi, will create a series of images.
+use KESMAnalysis.cli.makemp4.py to create a video or
+use imagemagick to create video from this frames
+convert -set delay 20 -loop 0 -quality 1000 -scale 100% *.png /home/pranathi/animExp.mpg
+"""
+
+
+def getFrames(pathThresh, pathSkel, totalTime, fps=24, totalRotation=360):
+    """
+    Return 3 vertex clique removed graph
+    Parameters
+    ----------
+    pathThresh : str
+        path of the .npy thresholded 3D Volume
+
+    pathSKel : str
+        path of the .npy skeleton 3D Volume
+
+    totalTime : integer
+        in seconds, duration of the video
+
+    fps : integer
+        frames per second, number of input frames per second
+
+    totalRotation : integer
+        angle in degrees frames should be captured in, integer between 0 and 360
+
+    Returns
+    -------
+    frames of png images in the same directory as pathThresh
+        mayavi scenes are saved as png images at different
+        angle of rotations as anim%i.png i is the ith frame
+
+    Notes
+    -----
+    threshold and skeletonized volume are overlapped,
+    thresholded volume's isosurface is transparent and
+    in grey and the skeletonized volume can be seen through
+    it and is in red
+    """
+    # total frames
```

Comment:
Add some assertions here, e.g. that 0 <= totalRotation <= 360, or you could do something like abs(totalRotation) % 360.
Also assert sensible values for other parameters. Does thresh and skel need to be the same size?